### PR TITLE
Add BracketsStartEndNamedSectionParser

### LINKED ISSUE #40: Named sections parsing
In code, there may be many names sections designated like this:

```
editable
[START section_name]
readonly
readonly
[END section_name]
editable
```

They will be used for read-only functionality, for initially collapsing blocks, and for hidden sections.
We need to parse them into `Code` object to get started on those features.

### PR DESCRIPTION
**Pull request by alexeyinkin**
Resolves #40

### CODE REVIEW

File: test/src/named_sections/parsers/brackets_start_end_test.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import 'package:code_text_field/src/named_sections/named_section.dart';
+import 'package:code_text_field/src/named_sections/parsers/brackets_start_end.dart';
+import 'package:code_text_field/src/single_line_comments/single_line_comment.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('BracketsStartEndNamedSectionParser', () {
+    const comments = [
+      //
```
Comment (nausharipov) [inline test/src/named_sections/parsers/brackets_start_end_test.dart:9]: is this intentional?

File: lib/src/named_sections/named_section.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import 'package:meta/meta.dart';
+
+@immutable
+class NamedSection {
+  /// Zero-based index of the line with the starting tag.
+  final int startLine;
+
+  /// Zero-based index of the line with the ending tag.
+  final int endLine;
+
```
Comment (nausharipov) [inline lib/src/named_sections/named_section.dart:10]: is the space between declarations needed?

File: test/src/named_sections/parsers/brackets_start_end_test.dart (hunk 1 above)
Comment (alexeyinkin) [reply test/src/named_sections/parsers/brackets_start_end_test.dart:9]: Yes, any comment in a list forces the dart formatter to preserve blank lines between list items.

File: lib/src/named_sections/named_section.dart (hunk 2 above)
Comment (alexeyinkin) [reply lib/src/named_sections/named_section.dart:10]: Blank lines separate doc comment + declaration pairs and improve doc comment clarity.
Declarations without doc comments are OK without blank lines between them.

**Pull request merged**